mods/ra/maps/soviet-09/soviet09.lua

```diff
@@ -0,0 +1,109 @@
+--[[
+   Copyright 2007-2020 The OpenRA Developers (see AUTHORS)
+   This file is part of OpenRA, which is free software. It is made
+   available to you under the terms of the GNU General Public License
+   as published by the Free Software Foundation, either version 3 of
+   the License, or (at your option) any later version. For more
+   information, see COPYING.
+]]
+
+TruckStops = { TruckStop1, TruckStop2, TruckStop3, TruckStop4, TruckStop5, TruckStop6, TruckStop7, TruckStop8 }
+MissionStartAttackUnits = { StartAttack1tnk1, StartAttack1tnk2, StartAttackArty1, StartAttackArty2, StartAttackArty3 }
+TruckEscape = { TruckEscape1, TruckEscape2, TruckEscape3, TruckEscape4, TruckEscape5, TruckEscapeWest }
+BackupRoute = { TruckEscape2, TruckEscape1, TruckEscapeEast }
+
+MissionStart = function()
+	Utils.Do(TruckStops, function(waypoint)
+		StolenTruck.Move(waypoint.Location)
+	end)
+
+	Trigger.AfterDelay(DateTime.Seconds(5), function()
+		Utils.Do(MissionStartAttackUnits, function(actor)
+			actor.AttackMove(DefaultCameraPosition.Location)
+		end)
+	end)
+
+	Trigger.AfterDelay(DateTime.Seconds(45), function()
+		Media.DisplayMessage("Commander, the truck has stopped at a nearby allied base.\nAllied radio intercepts say the truck has orders to flee the battlefield\nif any Soviet units approach the base.")
+	end)
+
+	Trigger.OnKilled(StolenTruck, function()
+		USSR.MarkCompletedObjective(DestroyTruck)
+		USSR.MarkCompletedObjective(DefendCommand)
+	end)
+
+	Trigger.OnKilled(CommandCenter, function()
+		USSR.MarkFailedObjective(DefendCommand)
+	end)
+end
+
+Trigger.OnEnteredProximityTrigger(TruckAlarm.CenterPosition, WDist.FromCells(13), function(actor, triggerflee)
+	if actor.Owner == USSR and not (actor.Type == "badr" or actor.Type == "u2") then
```

Comment:
Doesn't fully exclude spy planes or badgers, this will:

`	if actor.Owner == USSR and not (actor.Type == "badr" or actor.Type == "u2" or actor.Type == "camera.paradrop" or actor.Type == "camera.spyplane") then`

I don't know of a better way to exclude cameras from proxy triggers other than a new trait, or a re-usable proxy exclusion check in campaign-global.

Reply:
Fwiw, it would be better to use `~=` over `not ==`:
`if actor.Owner == USSR and actor.Type ~= "badr" and actor.Type ~= "u2" and actor.Type ~= "camera.paradrop" and actor.Type ~= "camera.spyplane" then`

I'm also not sure if we shouldn't have the truck start escaping when parabombs or paratroopers come in.

Reply:
I don't have parabombs available in this one. Paratroopers should trigger the truck when they jump out. I want to avoid a situation where the player drops paras on their side of the map and the plane passes over the trigger on the way out.

Reply:
> I don't have parabombs available in this one.

Interesting. What was the idea here, that it is too strong against the truck?

You're right about the planes passing, the better solution if we want it would be to drop the checks for the camera I guess.

Reply:
> What was the idea here, that it is too strong against the truck?

Yeah, then again I think everything is. Really no way to make this mission speed-run cheese proof.